cmd/kube-proxy/app/server_others.go

```diff
@@ -143,6 +143,9 @@ func newProxyServer(
 	nodeIP := net.ParseIP(config.BindAddress)
 	if nodeIP.IsUnspecified() {
 		nodeIP = utilnode.GetNodeIP(client, hostname)
+		if nodeIP == nil {
+			return nil, fmt.Errorf("unable to get node IP for hostname %s", hostname)
```

Comment:
I'm wondering if this is technically a breaking change. Previously if `nodeIP` is nil, kube-proxy defaults to [127.0.0.1 as the node IP](https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/iptables/proxier.go#L282-L286) but now kube-proxy always exits with error instead of defaulting to local host.

Reply:
Opened https://github.com/kubernetes/kubernetes/pull/84071 to preserve the original behavior but keep the retries for getting node object. PTAL :)